### PR TITLE
Review of methods page in reference

### PR DESCRIPTION
**Pull request by danimtb**
Added:
- Broken link references.
- References to methods starting with *method*
- Better param explanation with default values.
- Descrition of build()
- SystemPackageTool small section.
- Formatting

Removed:
- conan_info() method.

### CODE REVIEW

File: reference/conanfile/methods.rst [hunk 1]
```diff
@@ -27,7 +27,7 @@ you can directly get it from there:
             # You can also change branch, commit or whatever
             # self.run("cd hello && git checkout 2fe5...")
 
-This will work, as long as ``git`` is in your current path (so in Win you probably want to run things in msysgit, cmder, etc).
+This will work, as long as ``$ git`` is in your current path (so in Win you probably want to run things in msysgit, cmder, etc).
```
Comment (memsharded) [inline reference/conanfile/methods.rst]: This is not a command line, it is an executable. WIthout $

File: reference/conanfile/methods.rst [hunk 2]
```diff
@@ -71,26 +75,36 @@ You can also use **check_md5**, **check_sha1** and **check_sha256** from the **t
...
 -------
 
+This method is used to build the source code of the recipe using the desired commands. You can use your command line tools to invoke your
+build system or any of the build helpers provided with Conan.
+
+.. code-block:: python
+
+    def build(self):
+        cmake = CMake(self)
+        self.run("cmake . %s" % (cmake.command_line))
```
Comment (memsharded) [inline reference/conanfile/methods.rst]: Better use modern ``cmake.configure()`` syntax.

File: reference/conanfile/methods.rst [hunk 3]
```diff
@@ -100,63 +114,78 @@ Nothing special is required here. We can just launch the tests from the last com
 
     def build(self):
         cmake = CMake(self)
-        self.run("cmake . %s" % (cmake.command_line))
-        self.run("cmake --build . %s" % cmake.build_config)
+        cmake.configure()
+        cmake.build()
         # here you can run CTest, launch your binaries, etc
         self.run("ctest")
```
Comment (memsharded) [inline reference/conanfile/methods.rst]: Maybe replace this one with ``cmake.test()`` too

File: reference/conanfile/methods.rst (hunk 2 above)
Comment (danimtb) [reply reference/conanfile/methods.rst]: ``self.run()`` was left indentenly to show the ussage of both build helpers and normal command line instructions.

Comment (memsharded) [reply reference/conanfile/methods.rst]: Good, leave it.

**Pull request merged**